### ISSUE TITLE
[BUG] Shield of Tomorrow stack calculation mismatches

### ISSUE THREAD

**Issue by zeenk** [TLDRMissions/TLDRMissions#131]
Batch 2.

[Devtesting.zip](https://github.com/teelolws/TLDRMissions/files/8598215/Devtesting.zip)

**Comment by teelolws** [comment]
The file for 2277 has some calculations that don't make sense.
[2277.zip](https://github.com/teelolws/TLDRMissions/files/8598263/2277.zip)

    Kyrian Phalanx[1] applied Shield of Tomorrow to Kyrian Phalanx[1][2450HP].
    Kyrian Phalanx[2] applied Shield of Tomorrow to Kyrian Phalanx[1][2450HP].
    Invasive Lasher[10] shot Kyrian Phalanx[1][2299HP] for 151 damage.

So, Kyrian Phalanx has two buffs on it that reduce its damage take by 10% each. The base attack of Invasive Lasher is 190.

If we apply the buffs separately: 190 * 90% * 90% = 153.9, which is not the damage it dealt.

If we combine the buffs to be -20% damage taken, then the math: 190 * 80% = 152, is still not the damage it dealt.

Not sure what to do with this one.

**Comment by zeenk** [comment]
Weird, do you want more logs for this specific mission?

**Comment by teelolws** [comment]
Its not the mission, its Kyrian Phalanx, one of your troops. I need to do more Kyrian testing.

**Comment by zeenk** [comment]
oh I see

**Comment by teelolws** [comment]
Gormling Harrier[11] shot Kleia[0][2689HP] for 111 damage.

Baseattack 140, -20% from Shield of Tomorrow, should be 112.

**Comment by teelolws** [comment]
[TLDRMissions.zip](https://github.com/teelolws/TLDRMissions/files/8666864/TLDRMissions.zip)

**Comment by teelolws** [comment]
Okay theres my current theory, where it works for the 4-5 logs I checked it against:

- If a minion has both the "main" version of Shield of Tomorrow, and the "alt" version, then it takes an extra 1 less damage.
- If it just had two "alt" versions then it doesn't
- By "main" version, I mean the first part of the spell, where it casts on "itsself". By "alt", I mean the second part, where it casts the shield of two specific allies

Its weird but the logic seems to work for the logs I checked.

**Comment by teelolws** [comment]
"Voracious Collector[11] cast Icy Blast at Mikanikos[0][729HP] for 215  damage.", -- [43]

3 stacks of shield of tomorrow, none of them were the "main". Otherwise calculated as 216. Log found in https://github.com/teelolws/TLDRMissions/issues/195 2243

**Comment by teelolws** [comment]
![image](https://user-images.githubusercontent.com/49792789/169637602-005fe528-fd1a-463a-9ceb-df58d35bd97c.png)

**Comment by teelolws** [comment]
Well, I just solved Mikanikos' other mystery. Check this one, its interesting how Blizzards spaghetti code works. At least its not just me.

![image](https://user-images.githubusercontent.com/49792789/170854289-fa6c7848-31c4-438d-a3c5-7b8da0cd79e9.png)

The thorns from the Gormling Harrier healed the troop for 5782. How did Blizzard get this number?

With no buffs, the thorns would have done 118 damage. Thats its base attack.

But, the troop had Mikanikos' buff "Resilient Plumage" on it. This is supposed to reduce damage taken to zero, but I've found in other logs that it actually reduces damage by an unknown, big percentage. Because no other buff or debuff could negate it.

If, without any buffs/debuffs, the minion normally takes 100% of damage. Now apply the Plumage buff. Reduces damage taken by 5000%. 100% - 5000% = it takes -4900% of damage. Which normally just gets rounded to zero.

But for some reason, Thorns isn't capped at zero. It can go below zero, and heal its target!

That base attack earlier? 118 * 4900% = 5782. The amount Blizzard sent back.
